applications/callflow/src/cf_endpoint.erl

```diff
@@ -827,17 +833,63 @@ maybe_get_t38(Endpoint, Call) ->
              )
     end.
 
--spec maybe_build_failover(wh_json:object(), whapps_call:call()) -> api_object().
-maybe_build_failover(Endpoint, Call) ->
+-spec maybe_build_failover(wh_json:object(), clid(), whapps_call:call()) -> api_object().
+maybe_build_failover(Endpoint, Clid, Call) ->
     CallForward = wh_json:get_value(<<"call_forward">>, Endpoint),
     Number = wh_json:get_value(<<"number">>, CallForward),
     case wh_json:is_true(<<"failover">>, CallForward)
         andalso not wh_util:is_empty(Number)
     of
-        'false' -> 'undefined';
+        'false' -> maybe_build_push_failover(Endpoint, Clid, Call);
         'true' -> create_call_fwd_endpoint(Endpoint, wh_json:new(), Call)
     end.
 
+-spec maybe_build_push_failover(wh_json:object(), clid(), whapps_call:call()) -> api_object().
+maybe_build_push_failover(Endpoint, Clid, Call) ->
+    case wh_json:get_value(<<"push">>, Endpoint) of
+        'undefined' -> 'undefined';
+        Token -> build_push_failover(Endpoint, Clid, Token, Call)
+    end.
+
+-spec build_push_failover(wh_json:object(), clid(), binary(), whapps_call:call()) -> api_object().
+build_push_failover(Endpoint, Clid, PushObj, Call) ->
+    SIPJObj = wh_json:get_value(<<"sip">>, Endpoint),
+    ToUsername = get_to_username(SIPJObj),
+    ToRealm = cf_util:get_sip_realm(Endpoint, whapps_call:account_id(Call)),    
+    ToUser = <<ToUsername/binary, "@", ToRealm/binary>>,
+    Proxy = wh_json:get_value(<<"Token-Proxy">>, PushObj),
+    PushHeaders = wh_json:foldl(fun(K, V, Acc) -> 
+                                        wh_json:set_value(<<"X-KAZOO-PUSHER-", K/binary>>, V, Acc)  
+                                end, wh_json:new(), PushObj),
+    wh_json:from_list(
+      props:filter_empty(
+        [{<<"Invite-Format">>, <<"route">>}
+         ,{<<"To-User">>, ToUser}
+         ,{<<"To-Username">>, ToUsername}
+         ,{<<"To-Realm">>, ToRealm}
+         ,{<<"To-DID">>, get_to_did(Endpoint, Call)}
+         ,{<<"SIP-Transport">>, get_sip_transport(SIPJObj)}
+         ,{<<"Route">>, <<"sip:", ToUser/binary, ";fs_path='", Proxy/binary, "'">> }
```

Comment:
We'll need to figure out a way to specify a `Proxy` parameter instead of encoding it on the route here using `fs_path`. I don't think it should block this pull request but something to note (KAZOO-3303).